tensor/interpolateHypercube.cc

```diff
@@ -69,20 +71,28 @@ namespace civita
                 src.dimension.units!=dst.dimension.units) //TODO handle conversion between different units
               throw runtime_error("mismatch between unnamed dimensions");
             interimHC.xvectors.push_back(dst);
-            rotation[i]=i;
+            tmpRotation[i]=i;
           }
         else // find matching names dimension
           {
             auto dst=find_if(targetHC.begin(), targetHC.end(),
                              [&](const XVector& i){return i.name==src.name;});
             if (dst==targetHC.end())
-              throw runtime_error("no matching dimension found");
-            interimHC.xvectors.push_back(*dst);
-            rotation[dst-targetHC.begin()]=i;
+            {
+              // possible alternative when targetHC has no xvectors or undefined ones. for feature 147
+              interimHC.xvectors.push_back(src);
+              tmpRotation[i]=i;
+            }
+            else
+              {
+                interimHC.xvectors.push_back(*dst);
+                tmpRotation[dst-targetHC.begin()]=i;
+              }
           }
         strides.push_back(stride);
         stride*=targetHC[i].size();
       }
+    for (size_t i=0; i< tmpRotation.size(); i++) rotation[i]=tmpRotation[i];  
 #ifndef NDEBUG
     for (auto& i: rotation) assert(i<rank()); // check that no indices have been doubly assigned.
```

Comment:
For it to be a permutation, tmpRotation.size()==rank() must also be true, and this is cheaper than checking all rotation elements are < rank(). I think we need to throw in the case it is not a permutation.

Also line 95 can be replaced by for (auto& i: tmpRotation) rotation[i.first]=i.second, or alternatively, use push_back() (but need to remove line 59 first). It is cheaper to iterate over the elements of a map (which will in key order), than it is to look up the map elements each time around the loop (roughly O(N) instead of O(N log N).

Reply:
Done and done :)